Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -2350,6 +2354,100 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
+               let parentSymbol = parent.symbol
+            {
+                if parentToChildren[parentSymbol.identifier] != nil {
+                    parentToChildren[parentSymbol.identifier]?.append(child)
+                } else {
+                    parentToChildren[parentSymbol.identifier] = [child]
+                }
+            }
+        }
+        
+        // Create a dictionary of the names of all symbols and their documentation nodes.
+        // If there are any overloaded symbols, their names will collide.
+        for (_, children) in parentToChildren {
+            var overloadedSymbols = [String : [DocumentationNode]]()
+            for child in children {
+                
+                assert(
+                    child.semantic is Symbol,
+                    "The only nodes in the topic graph here should be symbols."
+                )
+                
+                guard let overloadName = getOverloadUnqiueName(for: child) else {
+                    // Symbol can't be overloaded
+                    continue
+                }
+                
+                if overloadedSymbols[overloadName] != nil {
+                    overloadedSymbols[overloadName]?.append(child)
+                } else {
+                    overloadedSymbols[overloadName] = [child]
+                }
+            }
+            
+            // Handle each group of overloads.
+            for (_, overloads) in overloadedSymbols {
+                guard overloads.count > 1 else {
+                    continue
+                }
+                
+                let overloadDeclarations = overloads.map {
+                    return Symbol.OtherDeclaration(declaration: ($0.semantic as! Symbol).declaration, identifier: $0.reference)
+                }
+                
+                // Create a documentation node for the generic overload. Currently this will contain the same content as one of the specific overloads
+                // to decrease requests made by the renderer.
+                let genericOverloadNode = overloads.first
+                (genericOverloadNode?.semantic as! Symbol).otherDeclarationsVariants = .init(swiftVariant: overloadDeclarations)
+                // TODO: Add genericOverloadNode to the topic graph.
+                
+                // Tell each symbol which other symbols overload it.
+                for (index, documentationNode) in overloads.indexed() {
+                    var otherOverloadDeclarations = overloadDeclarations
+                    otherOverloadDeclarations.remove(at: index)
+                    
+                    (documentationNode.semantic as! Symbol).otherDeclarationsVariants = .init(swiftVariant: otherOverloadDeclarations)
+                }
+            }
+        }
+    }
+    
+    /// Returns a name for an overload.
+    /// If two symbols have the same name, they are overloads of eachother.
+    private func getOverloadUnqiueName(for node: DocumentationNode) -> String? {
```

Comment:
Nit: Generally Swift API naming avoids the `get` prefix. Maybe something like this?

```suggestion
    private func uniqueName(forOverloadedSymbol node: DocumentationNode) -> String? {
```

The Swift API design guidelines are super helpful for this kind of thing: https://www.swift.org/documentation/api-design-guidelines/.